lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -812,33 +815,46 @@ function( $pseudo_selector ) use ( $selector ) {
 			$block_rules .= '}';
 		}
 
-		if ( static::ROOT_BLOCK_SELECTOR === $selector ) {
+		return $block_rules;
+	}
 
-			if ( $use_root_padding ) {
-				$block_rules .= '.wp-site-blocks { padding-top: var(--wp--style--root--padding-top); padding-bottom: var(--wp--style--root--padding-bottom); }';
-				$block_rules .= '.has-global-padding { padding-right: var(--wp--style--root--padding-right); padding-left: var(--wp--style--root--padding-left); }';
-				$block_rules .= '.has-global-padding > .alignfull { margin-right: calc(var(--wp--style--root--padding-right) * -1); margin-left: calc(var(--wp--style--root--padding-left) * -1); }';
-				$block_rules .= '.has-global-padding > .alignfull > :where([class*="wp-block-"]:not(.alignfull):not(.alignfull):not([class*="__"]),p,h1,h2,h3,h4,h5,h6,ul,ol) { padding-right: var(--wp--style--root--padding-right); padding-left: var(--wp--style--root--padding-left); }';
-			}
+	/**
+	 * Outputs the CSS for layout rules on the root.
+	 * This needs to be separate from get_styles_for_block,
+	 * so that it can load later and override block rules.
+	 *
+	 * @param string $selector The root node selector.
+	 * @param array $block_metadata The metadata for the root block.
+	 * @return string The additional root rules CSS.
+	 */
+	private function get_root_layout_rules( $selector, $block_metadata ) {
```

Comment:
This function appears to be working well for me in testing. Reading over the function, the logic looks a little split to me in the current state of the PR, given that the root layout content and wide size rules are still contained in `get_styles_for_block`. Would it be worth moving over the remaining lines from that function? (The block starting from around line 791 with `if ( static::ROOT_BLOCK_SELECTOR === $selector ) {`), here: https://github.com/WordPress/gutenberg/blob/a3bfa143adaaddcea6c8f54021217cb3fe4d7391/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php#L791-L816

I _think_ doing that would also resolve a subtle regression I just noticed that merged a while back. Ideally the `body { margin: 0; }` rule should be output before the spacing values for `margin` that can be set at the root `styles.spacing.margin` level in `theme.json` (originally fixed back in #36960 to resolve #36147).